### ISSUE TITLE
Full screen switch (using shared context) doesn't work anymore.

### ISSUE THREAD

**Issue by arampl** [glfw/glfw#635]
This code works with GLFW 3.1.1 and older versions, but GLFW 3.1.2 gives segmentation fault after switching to full screen mode:

```````c
#include <stdlib.h>
#include <stdio.h>
#include "glfw3.h"

GLFWmonitor *monitor;
GLFWvidmode *mode;
GLFWwindow  *window, *mainwindow;

int fullscreen = 0;

void recreate();

static void error(int error, const char* description)
{
    fputs(description, stderr);
}

void key_callback(GLFWwindow* window, int key, int scancode, int action, int mods)
{
   if(key == GLFW_KEY_F4 && action == GLFW_PRESS)
    {
        fullscreen = !fullscreen;
        recreate();
    }
}

void render()
{
    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
    glfwSwapBuffers(window);
}

void recreate()
{
    if(fullscreen)
    {
				glfwWindowHint(GLFW_RED_BITS, mode->redBits);
				glfwWindowHint(GLFW_GREEN_BITS, mode->greenBits);
				glfwWindowHint(GLFW_BLUE_BITS, mode->blueBits);
				glfwWindowHint(GLFW_REFRESH_RATE, mode->refreshRate);

				window = glfwCreateWindow(mode->width, mode->height, "title", monitor, mainwindow);
				glfwDestroyWindow(mainwindow);
				mainwindow = window;
				glfwMakeContextCurrent(mainwindow);
    }
    else
    {
				window = glfwCreateWindow(800, 600, "title", NULL, mainwindow);
				glfwDestroyWindow(mainwindow);
				mainwindow = window;
				glfwMakeContextCurrent(mainwindow);
				glfwShowWindow(mainwindow);
    }

    glfwSetKeyCallback(mainwindow, key_callback);
}

int main(int argc, char **argv)
{
    glfwSetErrorCallback(error);

    if(!glfwInit())
        return -1;

    monitor = glfwGetPrimaryMonitor();
    mode = (GLFWvidmode*)glfwGetVideoMode(monitor);

    glfwWindowHint(GLFW_VISIBLE, GL_FALSE);
    mainwindow = glfwCreateWindow(1, 1, "0", NULL, NULL);
    glfwMakeContextCurrent(mainwindow);

    recreate();

    while(!glfwWindowShouldClose(window))
    {
        render();
        glfwPollEvents();
    }

    glfwTerminate();
    return 0;
}

```````

**Comment by elmindreda** [comment]
Please see the section on [reentrancy](http://www.glfw.org/docs/latest/intro.html#reentrancy).

**Comment by arampl** [comment]
Thank you very much! Problem solved.

**Comment by elmindreda** [comment]
The proper solution to this, #43, will be included in 3.2.
